### PR TITLE
Make notification_substream_queued_bytes take a &self

### PR DESCRIPTION
**Pull request by tomaka**

### CODE REVIEW

File: src/libp2p/connection/yamux.rs [hunk 1]
```diff
@@ -895,6 +904,48 @@ pub struct Config {
...
+        self.id
+    }
+
+    /// Returns the user data associated to this substream.
+    pub fn user_data(&self) -> &T {
+        &self.substream.user_data
+    }
+
+    /// Returns the user data associated to this substream.
+    pub fn into_user_data(self) -> &'a T {
```
Comment (melekes) [inline src/libp2p/connection/yamux.rs:926]: _don't see much difference between this and above fn except `'a` lifetime._

**Review by melekes — APPROVED** [review]
👍

File: src/libp2p/connection/yamux.rs (hunk 1 above)
Comment (tomaka) [reply src/libp2p/connection/yamux.rs:926]: Yeah but they're fundamentally different.

These are methods on an object that borrows the connection.
`user_data(&self) -> &T` returns a `&T` that borrows the object that borrows the connection, while `into_user_data(self) -> &'a T` kills the object and returns a `&T` that borrows the connection.

Comment (melekes) [reply src/libp2p/connection/yamux.rs:926]: Oh, right. Thanks

**Pull request merged**